File: 백준/Silver/22858. 원상 복구 （small）/원상 복구 （small）.py
from sys import *
from collections import * 
from heapq import * 
from functools import *
from math import *
from itertools import *
import string

#setrecursionlimit(10**9)
input = stdin.readline

n , k = map(int,input().split())

s = list(map(int,input().split()))
d = list(map(int,input().split())) 

for _ in range (k):
    tmp = [0 for _ in range(n)]
    
    for i,v in enumerate(s):
        tmp[d[i]-1] = v  # 임시 배열에 해당 s 배열의 값을  이전 배치를 넣어준다.
    
    s = tmp

print(*s)
    

File: 백준/Silver/22858. 원상 복구 （small）/원상 복구 （small）.swift
import Foundation
let nk = readLine()!.split{$0 == " "}.map{Int($0)!}
let (n,k) = (nk[0],nk[1])

var s = readLine()!.split{$0 == " "}.map{Int($0)!}
let d = readLine()!.split{$0 == " "}.map{Int($0)!-1}


for _ in 0..<k {

    var tmp = s
    for i in 0..<n {
        tmp[d[i]] = s[i]
    }

    s = tmp
}

print(s.map{String($0)}.joined(separator: " "))

